packages/manager/cypress/e2e/core/parentChild/account-switching.spec.ts

```diff
@@ -0,0 +1,289 @@
+  profileFactory,
+} from '@src/factories';
+import { accountUserFactory } from '@src/factories/accountUsers';
+import { DateTime } from 'luxon';
+import {
+  mockCreateChildAccountToken,
+  mockCreateChildAccountTokenError,
+  mockGetAccount,
+  mockGetChildAccounts,
+  mockGetChildAccountsError,
+  mockGetUser,
+} from 'support/intercepts/account';
+import {
+  mockAppendFeatureFlags,
+  mockGetFeatureFlagClientstream,
+} from 'support/intercepts/feature-flags';
+import { mockAllApiRequests } from 'support/intercepts/general';
+import { mockGetProfile } from 'support/intercepts/profile';
+import { ui } from 'support/ui';
+import { makeFeatureFlagData } from 'support/util/feature-flags';
+import { assertLocalStorageValue } from 'support/util/local-storage';
+import { randomLabel, randomNumber, randomString } from 'support/util/random';
+
+/**
+ * Confirms expected username and company name are shown in user menu button and yields the button.
+ *
+ * @param username - Username to expect in user menu button.
+ * @param companyName - Company name to expect in user menu button.
+ *
+ * @returns Cypress chainable that yields the user menu button.
+ */
+const assertUserMenuButton = (username: string, companyName: string) => {
+  return ui.userMenuButton
+    .find()
+    .should('be.visible')
+    .within(() => {
+      cy.findByText(username).should('be.visible');
+      cy.findByText(companyName).should('be.visible');
+    });
+};
+
+/**
+ * Confirms that expected authentication values are set in Local Storage.
+ *
+ * @param token - Authentication token value to assert.
+ * @param expiry - Authentication expiry value to assert.
+ * @param scopes - Authentication scope value to assert.
+ */
+const assertAuthLocalStorage = (
+  token: string,
+  expiry: string,
+  scopes: string
+) => {
+  assertLocalStorageValue('authentication/token', token);
+  assertLocalStorageValue('authentication/expire', expiry);
+  assertLocalStorageValue('authentication/scopes', scopes);
+};
+
+const mockParentAccount = accountFactory.build({
+  company: 'Parent Company',
+});
+
+const mockParentProfile = profileFactory.build({
+  username: randomLabel(),
+  user_type: 'parent',
+});
+
+const mockParentUser = accountUserFactory.build({
+  username: mockParentProfile.username,
+  user_type: 'parent',
+});
+
+const mockChildAccount = accountFactory.build({
+  company: 'Child Company',
+});
+
+const mockChildAccountToken = appTokenFactory.build({
+  id: randomNumber(),
+  created: DateTime.now().toISO(),
+  expiry: DateTime.now().plus({ hours: 1 }).toISO(),
+  label: `${mockChildAccount.company}_proxy`,
+  scopes: '*',
+  token: randomString(32),
+  website: undefined,
+  thumbnail_url: undefined,
+});
+
+const mockErrorMessage = 'An unknown error has occurred.';
+
+describe('Parent/Child account switching', () => {
+  /*
+   * Tests to confirm that Parent account users can switch to Child accounts as expected.
+   */
+  describe('From Parent to Child', () => {
```

Comment:
I'm wondering here if it would be best for consistency to rename mentions of "child" to "proxy" in places like this throughout these tests because if we're being literal, a parent user can't switch to a `child` account, they switch to a `proxy` account which is their allocated user on the child's account.
```suggestion
  describe('From Parent to Proxy', () => {
```